opm/input/eclipse/Schedule/Well/Well.cpp

```diff
@@ -1837,6 +1837,21 @@ bool Well::handleWELSEGS(const DeckKeyword& keyword)
     return true;
 }
 
+void Well::addWellSegmentsFromLengthsAndDepths(const std::vector<std::pair<double, double>>& lengths_and_depths, double diameter, const KeywordLocation& location)
+{
+    if (this->segments == nullptr) {
```

Comment:
if (this->segments == nullptr || this->segments->empty()) . 

WELSEGS will add the top segment.